### PR TITLE
Add example module

### PR DESCRIPTION
**Pull request by fanhualta**
1. Add Kafka-IoTDB demo
2. Add RocketMQ-IoTDB demo
3. Add readme

### CODE REVIEW

File: example/readme.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+Storage groups are used to let users define how to organize and isolate different time series data on disk. Time series belonging to the same storage group will be continuously written to the same file in the corresponding folder. The file may be closed due to user commands or system policies, and hence the data coming next from these sensors will be stored in a new file in the same folder. Time series belonging to different storage groups are stored in different folders.
+##Connector
+> note:In this sample program, there are some update operations for historical data, so it is necessary to ensure the sequential transmission and consumption of data via RocketMQ. If there is no update operation in use, then there is no need to guarantee the order of data. IoTDB will process these data which may be disorderly.
+
+###Producer
+Producers insert IoTDB insert statements into partitions according to devices, ensuring that the same device's data is inserted or updated in the same MessageQueue.
+###Consumer 
+1. At startup, consumer client first creates the JDBC connection and check whether storage groups and timeseries are created in IoTDB. If it is not created, create it.
```
Comment (qiaojialin) [inline example/readme.md]:
```suggestion
1. At startup, the consumer client first creates a JDBC connection and check whether the storage groups and timeseries are created in IoTDB. If not, create it.  
```

File: example/src/main/java/org/apache/iotdb/example/Utils.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iotdb.example;
+
+/**
+ * Created by litianan on 2019/1/22.
```
Comment (qiaojialin) [inline example/src/main/java/org/apache/iotdb/example/Utils.java]: remove this...

File: example/src/main/java/org/apache/iotdb/example/kafkademo/Constant.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  public static final String TOPIC = "Kafka-Test";
+  public static final int CONSUMER_THREAD_NUM = 5;
+  public static final String IOTDB_CONNECTION_URL = Config.IOTDB_URL_PREFIX + "localhost:6667/";
+  public static final String IOTDB_CONNECTION_USER = "root";
+  public static final String IOTDB_CONNECTION_PASSWORD = "root";
+  public static final String STORAGE_GROUP = "root.vehicle";
+  public static final String[] ALL_TIMESERIES = {"root.vehicle.device.sensor1",
+      "root.vehicle.device.sensor2", "root.vehicle.device.sensor3", "root.vehicle.device.sensor4"};
+  public static final String[] ALL_DATA = {
+      "sensor1,2017/10/24 19:30:00,60 61 62",
```
Comment (qiaojialin) [inline example/src/main/java/org/apache/iotdb/example/kafkademo/Constant.java]: This format is a little strange. Why are there 3 values at a time point?
The format we usually met is: Timestamp, deviceId, sensor1, sensor2, sensor3

File: example/src/main/java/org/apache/iotdb/example/Utils.java (hunk 2 above)
Comment (fanhualta) [reply example/src/main/java/org/apache/iotdb/example/Utils.java]: fixed

File: example/src/main/java/org/apache/iotdb/example/kafkademo/Constant.java (hunk 3 above)
Comment (fanhualta) [reply example/src/main/java/org/apache/iotdb/example/kafkademo/Constant.java]: fixed

**Comment by jixuan1989** [comment]
I suggest you organize the examples into two different projects: examples/kafka; examples/rocketMQ

**Comment by jixuan1989** [comment]
I modify the pom.xml files to set example as a module.

File: example/pom.xml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>org.apache.iotdb</groupId>
+                <artifactId>iotdb-jdbc</artifactId>
+                <version>${project.version}</version>
+            </dependency>
+        </dependencies>
+    </dependencyManagement>
+    <dependencies>
```
Comment (MyXOF) [inline example/pom.xml]: why add iotdb-jdbc to dependencyManagement and dependencies

Comment (fanhualta) [reply example/pom.xml]: Unify dependencies in sub-project

**Pull request merged**